### PR TITLE
[Feature] Support explain analyze

### PR DESCRIPTION
**Pull request by liuyehcf**
## Background

Currently, we've already had profile and profile visualization, we can get the bottleneck of a given query based on the profile visualization feature, but unfortunately this feature is only opend for enterprise edition, it's not friendly for the users of the community edition to understand the profile content.

Therefore, there is an urgent need for a tool that can provide a concise and comprehensible analysis of the runtime execution.

## How

Syntax is `explain analyze <original sql>`

## Limitation

1. Only support query, which starts with select. 
2. If the query fails, the analyze also fails.

## Example

image

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 35 (54.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|2|00.00%|[1759, 1767]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|0|1|00.00%|[133]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|0|3|00.00%|[347, 348, 350]|
|:large_blue_circle:|com/starrocks/common/util/Counter.java|0|2|00.00%|[60, 61]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|6|12|50.00%|[553, 554, 555, 826, 827, 828]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|4|75.00%|[139]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|5|6|83.33%|[213]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 353 (06.80%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|2|00.00%|[1759, 1767]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|0|1|00.00%|[133]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|0|3|00.00%|[347, 348, 350]|
|:large_blue_circle:|com/starrocks/sql/ExplainAnalyzer.java|0|313|00.00%|[54, 57, 60, 61, 62, 63, 64, 68, 69, 70, 74, 75, 91, 92, 93, 95, 98, 99, 100, 101, 104, 106, 107, 109, 110, 113, 116, 119, 120, 121, 122, 123, 124, 127, 128, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 144, 148, 149, 151, 152, 154, 156, 157, 158, 159, 160, 162, 164, 167, 169, 170, 171, 172, 173, 174, 175, 176, 178, 179, 182, 184, 186, 187, 191, 192, 193, 194, 195, 196, 197, 199, 200, 201, 203, 204, 207, 208, 209, 210, 212, 215, 216, 217, 218, 219, 220, 221, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 237, 239, 241, 243, 244, 247, 248, 249, 250, 252, 253, 255, 256, 258, 261, 262, 264, 265, 267, 269, 270, 272, 273, 274, 278, 279, 283, 284, 285, 286, 287, 288, 289, 290, 293, 294, 295, 296, 297, 298, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 323, 324, 325, 326, 329, 330, 331, 332, 333, 335, 336, 338, 340, 341, 343, 344, 347, 348, 349, 350, 351, 352, 353, 355, 356, 359, 360, 362, 364, 369, 370, 371, 372, 374, 375, 376, 377, 378, 379, 381, 382, 384, 385, 387, 388, 391, 392, 394, 398, 399, 400, 402, 406, 407, 410, 411, 414, 415, 416, 418, 419, 420, 421, 422, 424, 427, 428, 431, 432, 435, 436, 457, 458, 459, 460, 461, 464, 465, 466, 469, 470, 472, 473, 474, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 491, 492, 504, 505, 506, 507, 508, 509, 510, 513, 514, 515, 516, 518, 519, 520, 524, 525, 526, 527, 530, 531, 532, 533, 534, 538, 539, 541, 543, 544, 546, 547, 548, 549, 550, 551, 553, 555, 559, 563, 564, 568, 569, 570, 572, 573, 575]|
|:large_blue_circle:|com/starrocks/common/util/Counter.java|0|2|00.00%|[60, 61]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|6|12|50.00%|[553, 554, 555, 826, 827, 828]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|4|75.00%|[139]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|5|6|83.33%|[213]|
|:large_blue_circle:|com/starrocks/planner/AnalyticEvalNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 353 (06.80%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|2|00.00%|[1759, 1767]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|0|1|00.00%|[133]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|0|3|00.00%|[347, 348, 350]|
|:large_blue_circle:|com/starrocks/sql/ExplainAnalyzer.java|0|313|00.00%|[54, 57, 60, 61, 62, 63, 64, 68, 69, 70, 74, 75, 91, 92, 93, 95, 98, 99, 100, 101, 104, 106, 107, 109, 110, 113, 116, 119, 120, 121, 122, 123, 124, 127, 128, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 144, 148, 149, 151, 152, 154, 156, 157, 158, 159, 160, 162, 164, 167, 169, 170, 171, 172, 173, 174, 175, 176, 178, 179, 182, 184, 186, 187, 191, 192, 193, 194, 195, 196, 197, 199, 200, 201, 203, 204, 207, 208, 209, 210, 212, 215, 216, 217, 218, 219, 220, 221, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 237, 239, 241, 243, 244, 247, 248, 249, 250, 252, 253, 255, 256, 258, 261, 262, 264, 265, 267, 269, 270, 272, 273, 274, 278, 279, 283, 284, 285, 286, 287, 288, 289, 290, 293, 294, 295, 296, 297, 298, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 323, 324, 325, 326, 329, 330, 331, 332, 333, 335, 336, 338, 340, 341, 343, 344, 347, 348, 349, 350, 351, 352, 353, 355, 356, 359, 360, 362, 364, 369, 370, 371, 372, 374, 375, 376, 377, 378, 379, 381, 382, 384, 385, 387, 388, 391, 392, 394, 398, 399, 400, 402, 406, 407, 410, 411, 414, 415, 416, 418, 419, 420, 421, 422, 424, 427, 428, 431, 432, 435, 436, 457, 458, 459, 460, 461, 464, 465, 466, 469, 470, 472, 473, 474, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 491, 492, 504, 505, 506, 507, 508, 509, 510, 513, 514, 515, 516, 518, 519, 520, 524, 525, 526, 527, 530, 531, 532, 533, 534, 538, 539, 541, 543, 544, 546, 547, 548, 549, 550, 551, 553, 555, 559, 563, 564, 568, 569, 570, 572, 573, 575]|
|:large_blue_circle:|com/starrocks/common/util/Counter.java|0|2|00.00%|[60, 61]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|6|12|50.00%|[553, 554, 555, 826, 827, 828]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|4|75.00%|[139]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|5|6|83.33%|[213]|
|:large_blue_circle:|com/starrocks/planner/AnalyticEvalNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 388 (05.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|2|00.00%|[1759, 1767]|
|:large_blue_circle:|com/starrocks/planner/AnalyticEvalNode.java|0|1|00.00%|[112]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|0|1|00.00%|[133]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|0|3|00.00%|[347, 348, 350]|
|:large_blue_circle:|com/starrocks/planner/JoinNode.java|0|1|00.00%|[331]|
|:large_blue_circle:|com/starrocks/sql/ExplainAnalyzer.java|0|347|00.00%|[60, 63, 66, 67, 68, 69, 70, 74, 75, 76, 80, 81, 97, 98, 99, 101, 104, 105, 106, 107, 110, 112, 113, 115, 116, 119, 122, 125, 126, 127, 128, 129, 130, 133, 134, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 150, 154, 155, 157, 158, 160, 162, 163, 164, 165, 166, 168, 170, 173, 175, 176, 177, 178, 179, 180, 181, 182, 184, 185, 188, 190, 192, 193, 197, 198, 199, 200, 201, 202, 203, 205, 206, 207, 209, 210, 213, 214, 215, 216, 218, 221, 222, 223, 224, 225, 226, 227, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 243, 245, 247, 249, 250, 253, 254, 255, 256, 258, 259, 261, 262, 264, 267, 268, 270, 271, 273, 275, 276, 278, 279, 280, 284, 285, 289, 290, 291, 292, 293, 294, 295, 296, 299, 300, 301, 302, 303, 304, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 329, 330, 331, 332, 336, 337, 339, 341, 342, 343, 346, 347, 350, 351, 352, 353, 354, 356, 357, 358, 359, 361, 362, 363, 364, 366, 367, 368, 369, 370, 371, 373, 374, 375, 376, 378, 379, 380, 381, 382, 384, 385, 386, 387, 388, 389, 390, 391, 392, 395, 398, 399, 400, 401, 402, 403, 404, 406, 407, 410, 411, 413, 415, 420, 421, 422, 423, 425, 426, 427, 428, 429, 430, 432, 433, 435, 436, 438, 439, 442, 443, 445, 449, 450, 451, 453, 457, 458, 461, 462, 465, 466, 467, 469, 470, 471, 472, 473, 475, 478, 479, 482, 483, 486, 487, 508, 509, 510, 511, 512, 515, 516, 517, 520, 521, 523, 524, 525, 527, 528, 529, 530, 531, 532, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 555, 556, 557, 558, 559, 560, 561, 564, 565, 566, 567, 569, 570, 571, 575, 576, 577, 578, 581, 582, 583, 584, 585, 589, 590, 592, 594, 595, 597, 598, 599, 600, 601, 602, 604, 606, 610, 614, 615, 619, 620, 621, 623, 624, 626]|
|:large_blue_circle:|com/starrocks/common/util/Counter.java|0|2|00.00%|[60, 61]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|6|12|50.00%|[553, 554, 555, 826, 827, 828]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|4|75.00%|[139]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|5|6|83.33%|[213]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java [hunk 1]
```diff
@@ -343,7 +343,14 @@ public String toString() {
         return formater.format(this);
     }
 
-    private String printCounter(long value, TUnit type) {
+    public static String printCounter(Counter counter) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:346]: What's the purpose of `explain analyze`? It's better to add a commit message

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java (hunk 1 above)
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:346]: This pr is under developmenet now, I've changed the title, and I'll add description after finishing this pr.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 586 (03.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|0|4|00.00%|[1801, 1802, 1809, 1810]|
|:large_blue_circle:|com/starrocks/planner/AnalyticEvalNode.java|0|1|00.00%|[112]|
|:large_blue_circle:|com/starrocks/planner/AggregationNode.java|0|1|00.00%|[133]|
|:large_blue_circle:|com/starrocks/planner/SortNode.java|0|1|00.00%|[120]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|0|3|00.00%|[347, 348, 350]|
|:large_blue_circle:|com/starrocks/planner/JoinNode.java|0|1|00.00%|[331]|
|:large_blue_circle:|com/starrocks/sql/ExplainAnalyzer.java|0|537|00.00%|[71, 74, 84, 85, 86, 87, 88, 92, 93, 95, 96, 97, 101, 102, 105, 106, 107, 109, 110, 112, 113, 115, 116, 117, 126, 127, 128, 129, 130, 131, 133, 138, 139, 140, 141, 144, 146, 147, 149, 150, 153, 156, 159, 160, 161, 162, 163, 164, 167, 168, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 184, 188, 189, 191, 192, 194, 196, 197, 198, 199, 200, 202, 206, 207, 208, 209, 210, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 225, 226, 227, 229, 230, 232, 233, 234, 235, 236, 238, 239, 242, 243, 244, 245, 246, 248, 250, 251, 252, 253, 254, 256, 258, 260, 261, 262, 264, 265, 266, 267, 268, 270, 272, 274, 277, 278, 280, 281, 282, 284, 285, 288, 290, 292, 293, 297, 298, 299, 300, 301, 302, 303, 305, 306, 307, 308, 309, 311, 312, 313, 315, 316, 319, 320, 321, 322, 324, 327, 328, 329, 330, 331, 332, 333, 334, 336, 337, 340, 341, 343, 344, 345, 346, 347, 348, 349, 351, 352, 353, 354, 355, 356, 358, 359, 360, 361, 362, 363, 366, 368, 370, 372, 373, 376, 377, 378, 380, 381, 382, 383, 384, 387, 388, 389, 391, 392, 394, 397, 398, 400, 401, 403, 405, 406, 408, 410, 411, 412, 413, 414, 416, 420, 421, 425, 426, 427, 428, 429, 430, 431, 432, 435, 436, 437, 438, 439, 440, 443, 444, 448, 449, 450, 451, 452, 453, 454, 456, 457, 458, 461, 462, 466, 467, 469, 471, 472, 473, 476, 477, 478, 479, 481, 485, 486, 487, 488, 490, 491, 495, 496, 499, 500, 501, 502, 503, 505, 506, 508, 509, 511, 512, 513, 514, 516, 517, 519, 520, 521, 523, 524, 525, 526, 527, 528, 529, 530, 531, 532, 535, 536, 537, 538, 539, 540, 541, 542, 544, 545, 546, 547, 548, 549, 550, 551, 552, 553, 555, 556, 557, 558, 560, 561, 562, 563, 564, 566, 567, 569, 570, 571, 572, 573, 576, 579, 580, 581, 582, 583, 584, 585, 586, 587, 589, 590, 592, 593, 595, 596, 598, 599, 602, 603, 604, 605, 606, 607, 610, 611, 613, 615, 619, 620, 621, 623, 624, 625, 626, 627, 630, 631, 633, 635, 643, 644, 645, 646, 647, 648, 649, 651, 652, 653, 655, 660, 661, 662, 663, 665, 666, 667, 668, 669, 670, 672, 673, 675, 676, 678, 679, 682, 683, 685, 689, 690, 691, 693, 697, 698, 701, 702, 705, 706, 707, 709, 710, 711, 712, 713, 716, 717, 718, 719, 720, 721, 723, 726, 727, 728, 731, 732, 735, 736, 739, 740, 743, 744, 747, 748, 766, 770, 771, 772, 773, 774, 777, 778, 779, 782, 783, 784, 785, 786, 787, 788, 789, 791, 792, 793, 794, 797, 798, 801, 802, 803, 804, 806, 808, 809, 810, 811, 812, 813, 814, 815, 816, 817, 818, 819, 820, 821, 822, 823, 824, 825, 828, 829, 830, 831, 833, 844, 845, 846, 847, 848, 849, 850, 853, 854, 855, 856, 858, 859, 860, 861, 862, 867, 868, 869, 870, 872, 873, 874, 875, 876, 880, 881, 883, 885, 886, 888, 889, 890, 891, 892, 893, 895, 897, 901, 905, 906, 910, 911, 912, 914, 915, 917]|
|:large_blue_circle:|com/starrocks/common/util/Counter.java|0|2|00.00%|[60, 61]|
|:large_blue_circle:|com/starrocks/planner/ProjectNode.java|0|2|00.00%|[58, 62]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|5|11|45.45%|[212, 1819, 1826, 1827, 1828, 1829]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|7|11|63.64%|[575, 847, 848, 918]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|3|4|75.00%|[139]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -2258,7 +2258,7 @@ restoreTableDesc
     ;
 
 explainDesc
-    : (DESC | DESCRIBE | EXPLAIN) (LOGICAL | VERBOSE | COSTS)?
+    : (DESC | DESCRIBE | EXPLAIN) (LOGICAL | ANALYZE |VERBOSE | COSTS)?
```
Comment (fzhedu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:2261]: only supprt explan analyze select stmt, how about update or other stmt?

**Review by fzhedu — COMMENTED** [review]
can it supported for Trino sql?

**Comment by liuyehcf** [comment]
> can it supported for Trino sql?

Emm, what do you mean by `Trino sql`?

**Comment by liuyehcf** [comment]
This feature cannot be covered by unit test. I'll add tests in `sql test` later after the standard data set(like tpch, tpcds) is supported.

**Review by satanson — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Pull request merged**